api/specs.proto

```diff
@@ -128,6 +128,21 @@ message TaskSpec {
 	// using the same reconciliation-based mechanism that performs rolling
 	// updates.
 	uint64 force_update = 9;
+
+	// ResourceReferences provides a generic way to specify resources that
+	// are used by this task, and should be sent down to agents along with
+	// the task. Inside the runtime field there may be more specific
+	// information about how to use the resource, but ResourceReferences
+	// establishes the relationship at the store level, and instructs the
+	// dispatcher to send the related objects.
+	//
+	// ResourceReferences is a list of ResourceReferences used by the task.
+	repeated ResourceReference resource_references = 11;
```

Comment:
nit: It might be a good idea to add `[(gogoproto.nullable) = false]` here, just before the semicolon. This causes the code generator to create this field as `[]ResourceReference` instead of `[]*ResourceReference`. The pointer indirection isn't really useful here, so I find it a little cleaner and more efficient to avoid it. It also makes it a little easier to reason about whether there's a risk of a nil pointer dereference in any particular case.